draft-gpew-priv-ppm.md

```diff
@@ -900,37 +900,70 @@ The named parameters are:
 * `batch_interval`, the request's batch interval.
 * `agg_param`, an aggregation parameter for the VDAF being executed.
 
-Depending on the VDAF scheme and how the leader is configured, the collect
-request may cause the leader to send a series of aggregate requests to the
-helpers in order to compute their share of the aggregate. Alternately, the
-leader may already have made these requests and can respond immediately. In
-either case it responds to the collector's request as follows.
+Depending on the VDAF scheme and how the leader is configured, the leader and
+helper may already have prepared all the reports falling within `batch_interval`
+and be ready to return the aggregate shares right away, but this cannot be
+guaranteed and in some VDAFs, it may not be possible to begin preparing inputs
+until the collector provides the aggregation parameter in the `CollectReq`. For
+those reasons, collect requests are handled asynchronously.
+
+Upon receipt of a `CollectReq`, the leader begins by checking that the request
+meets the requirements of the batch parameters using the procedure in
+{{batch-parameter-validation}}. If so, it immediately sends the collector a
+response with HTTP status 202 Accepted. The body consists of a `CollectResp`:
 
-It begins by checking that the request meets the requirements of the batch
-parameters using the procedure in {{batch-parameter-validation}}. If so, it
-obtains the helper's encrypted aggregate share for the batch interval by sending
-an AggregateShareReq to the helper as described in {{aggregate-share-request}}.
-(This request may too have been made in advance.)
-
-Next, the leader computes its own aggregate share by aggregating all of the
-verified output shares that fall within the batch interval. Finally, it responds
-with HTTP status 200 and a body consisting of a CollectResp message:
-
-[OPEN ISSUE: What happens if this all takes a really long time.]
+~~~
+struct {
+  Url collect_job;
+} CollectResp;
+~~~
 
-[TODO: Decide if and how the collector's request is authenticated.]
+`CollectResp.collect_job` is a URL identifying the collect job that can be
+polled by the collector. It SHOULD contain an unpredictable component to
```

Comment:
This SHOULD seems odd. If this is a real attack then there should be a MUST requiring high entropy.

I'm also not clear what an "enumeration" attack is. Presumably the leader has an access control mechanism, so what does it matter if I try /results/1 and get permission denied? And if it's supposed to prevent me knowing how many jobs there are, then I think this needs to be entirely unpredictable.

Reply:
You're right, it makes more sense to assume this is authenticated. I'll delete this text.